rest/server_context.go

```diff
@@ -1008,11 +1006,18 @@ func (sc *ServerContext) logStats() error {
 
 func AddGoRuntimeStats() {
 
-	statsResourceUtilizationVar := base.GlobalStats.Get(base.StatsGroupKeyResourceUtilization)
-	statsResourceUtilization := statsResourceUtilizationVar.(*expvar.Map)
+	statsResourceUtilization := base.StatsResourceUtilization()
 
 	// Num goroutines
-	statsResourceUtilization.Set("num_goroutines", base.ExpvarIntVal(runtime.NumGoroutine()))
+	statsResourceUtilization.Set(base.StatKeyNumGoroutines, base.ExpvarIntVal(runtime.NumGoroutine()))
+
+	// Goroutines high watermark
+	numGoroutines := uint64(runtime.NumGoroutine())
+	if numGoroutines > base.MaxGoroutinesSeen {
+		atomic.StoreUint64(&base.MaxGoroutinesSeen, numGoroutines)
+	}
+	maxGoroutinesSeen := atomic.LoadUint64(&base.MaxGoroutinesSeen)
```

Comment:
The use the atomic int here doesn't look correct:
- line 1016 is reading the value directly, instead of using load.
- line 1017 is using Store instead of CompareAndSwap, so doesn't handle the case where the value is updated since the read on line 1016
- We don't need to update the expvar if the value hasn't changed

Reply:
Thanks for catching that.  Fixed + added test.

I'm intentionally ignoring CAS errors, since it doesn't seem worth it (nor does it seem likely to happen).